target/linux/layerscape/base-files/lib/upgrade/platform.sh

```diff
@@ -60,7 +60,6 @@ platform_do_upgrade_sdboot() {
 
 	if [ -n "$diff" ]; then
 		dd if="$1" of="/dev/$diskdev" bs=1024 count=4 > /dev/null 2>&1
-		dd if="$1" of="$diskdev" bs=1024 skip=4 seek=16384 > /dev/null 2>&1
```

Comment:
This line should be fixed as below instead of removing. The line is writing kernel+rootfs for non-partition-saving case.
dd if="$1" of="/dev/$diskdev" bs=1024 skip=4 seek=16384 > /dev/null 2>&1

Reply:
It shouldn't be here. In this moment kernel is written on every case:
https://github.com/openwrt/openwrt/blob/ffc3a1ea8bc0e9306f05bd08f62fd80e5e732a07/target/linux/layerscape/base-files/lib/upgrade/platform.sh#L75

If we keep this line, kernel will be written two times for non-partition-saving case.

Reply:
No. There is a return in "if [ -n "$diff" ]" section.

Reply:
Sorry, my bad.